caravel/assets/javascripts/SqlLab/components/QuerySearch.jsx

```diff
@@ -1,67 +1,87 @@
+const $ = window.$ = require('jquery');
 import React from 'react';
-import Select from 'react-select';
-import { Button } from 'react-bootstrap';
 
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import * as Actions from '../actions';
-import QueryTable from './QueryTable';
+import QuerySearchTable from './QuerySearchTable';
+import QuerySearchBar from './QuerySearchBar';
 
 class QuerySearch extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      queryText: '',
+      queriesArray: [],
     };
   }
-  changeQueryText(value) {
-    this.setState({ queryText: value });
+  componentWillMount() {
+    this.refreshQueries(this.props);
+  }
+  componentWillReceiveProps(nextProps) {
+    this.refreshQueries(nextProps);
+  }
+  getQueryState(val) {
```

Comment:
getQueryState(val) {
switch (val) {
       case 1:
         return 'success';
etc  will be a bit less of code